File: Karthik/9-9_to_10-9/arrayzero.py

def minimumOperations(nums):
    """
    :type nums: List[int]
    :rtype: int
    """
    selected=minother(nums)
    count=0
    while(nums.count(0)!=len(nums)):
        subract=nums[selected]
        for i in range(len(nums)):
            if nums[i]!=0:
                nums[i]=nums[i]-subract
        if nums[selected]==0:
            selected=minother(nums)
        count+=1
    return count

def minother(nums):
    min=nums.index(max(nums))
    for i in range(len(nums)):
        if nums[i]!=0 and nums[i]<nums[min]:
            min=i
    return min

minimumOperations([1,5,0,3,5])
        

File: Srija/Week 1/arrayzero.cpp
class Solution {
public:
    int minimumOperations(vector<int>& nums) {
        int c=0;
        int flag = 1;
        while(flag){
            int m = 100;
            flag = 0;
            for(int i=0; i<nums.size(); i++){
                if(nums[i] < m && nums[i]!=0){
                    m = nums[i];
                }
                if(nums[i] != 0){
                    flag = 1;
                }
            }
            cout<<m<<" ";
            for(int i=0; i<nums.size(); i++){
                if(nums[i]!=0){
                    nums[i] -= m;
                }
            }
            c++;           
            
        }
       return (c-1);
    }
};